modular_skyrat/modules/primitive_fun/code/cutting_board.dm

```diff
@@ -0,0 +1,121 @@
+	. += span_notice("It can be (un)secured with <b>Right Click</b>")
+	. += span_notice("You can make it drop its item with <b>Alt Click</b>")
+	if(length(contents))
+		. += span_notice("It has [contents[1]] sitting on it.")
+
+/obj/item/cutting_board/Destroy()
+	drop_everything_contained()
+	return ..()
+
+/obj/item/cutting_board/AltClick(mob/user)
+	if(!length(contents))
+		balloon_alert(user, "nothing on board")
+		return
+	drop_everything_contained()
+	balloon_alert(user, "cleared board")
+
+///Drops all contents at the turf of the item
+/obj/item/cutting_board/proc/drop_everything_contained()
+	if(!length(contents))
+		return
+	for(var/obj/target_item as anything in contents)
+		target_item.forceMove(get_turf(src))
+
+/obj/item/cutting_board/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!can_interact(user) || !user.canUseTopic(src, be_close = TRUE))
+		return
+	set_anchored(!anchored)
+	balloon_alert_to_viewers(anchored ? "secured" : "unsecured")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+///Takes the given obj (processed thing) and gets its results from the recipe list, spawning the results and deleting the original obj
+/obj/item/cutting_board/proc/process_food(datum/food_processor_process/recipe, obj/processed_thing)
+	if(!recipe.output || !loc || QDELETED(src))
+		return
+	var/food_multiplier = recipe.food_multiplier
+	for(var/i in 1 to food_multiplier)
+		var/obj/new_food_item = new recipe.output(drop_location())
+		new_food_item.pixel_x = rand(-6, 6)
+		new_food_item.pixel_y = rand(-6, 6)
+		if(!processed_thing.reagents) //backup in case we really fuck up
+			continue
+		processed_thing.reagents.copy_to(new_food_item, processed_thing.reagents.total_volume, multiplier = 1 / food_multiplier)
+	qdel(processed_thing)
+	update_appearance()
+
+/obj/item/cutting_board/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(attacking_item.tool_behaviour == TOOL_KNIFE)
+		if(!length(contents))
+			balloon_alert(user, "nothing to process")
+			return
+		var/datum/food_processor_process/item_process_recipe = GET_RECIPE(contents[1])
+		if(!item_process_recipe)
+			log_admin("DEBUG: [src] (cutting board item) just tried to process [contents[1]] but wasn't able to get a recipe somehow, this should not be able to happen.")
+			return
+		playsound(src, 'sound/effects/butcher.ogg', 50, TRUE)
+		balloon_alert_to_viewers("cutting...")
+		if(!do_after(user, 3 SECONDS, target = src))
+			balloon_alert_to_viewers("stopped cutting")
+			return
+		process_food(item_process_recipe, contents[1])
+		return
+
+	var/datum/food_processor_process/gotten_recipe = GET_RECIPE(attacking_item)
+	if(!gotten_recipe)
+		balloon_alert(user, "can't process [attacking_item]")
+		return
+	if(length(contents))
+		balloon_alert(user, "board is full")
+		return
+	attacking_item.forceMove(src)
+	balloon_alert(user, "placed [attacking_item] on board")
+	update_appearance()
```

Comment:
I know I like early returns, but I think for the sake of having ..(), you need to make it if(gotten_recipe) and then indent everything. Afterwards, return and then unindent a new return ..()